libraries/ms-drivers/inc/voltage_regulator.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+// Driver for the voltage regulator
+// Use this 5V regulator driver when the hardware needs 5v power.
+// Requires interrupts, soft timers, and gpio to be initialized.
+
+// voltage_regulator_init initializes the regulator with the
+// provided settings and starts the softimer.
+
+#include "gpio.h"
+#include "soft_timer.h"
+
+typedef enum {
+  VOLTAGE_REGULATOR_ERROR_ON_WHEN_SHOULD_BE_OFF,
+  VOLTAGE_REGULATOR_ERROR_OFF_WHEN_SHOULD_BE_ON,
+  NUM_VOLTAGE_REGULATOR_ERRORS
+} VoltageRegulatorError;
+
+typedef void (*VoltageRegulatorErrorCallback)(VoltageRegulatorError error, void *context);
+
+typedef struct {
+  bool regulator_on;
+  GpioAddress enable_pin;
+  GpioAddress monitor_pin;
+  SoftTimerId timer_id;
+  uint32_t timer_callback_delay;
+  VoltageRegulatorErrorCallback error_callback;
+  void *error_callback_context;
+} VoltageRegulatorStorage;
+
+typedef struct {
+  GpioAddress enable_pin;
+  GpioAddress monitor_pin;
+  uint32_t timer_callback_delay;
```

Comment:
Please change this name to `timer_callback_delay_ms` so that people at least know what the unit is.